include/libtorrent/block_cache.hpp

```diff
@@ -337,7 +332,7 @@ namespace libtorrent
 		struct hash_value
 		{
 			std::size_t operator()(cached_piece_entry const& p) const
-			{ return std::size_t(p.storage.get()) + std::size_t(p.piece); }
+			{ return std::size_t(p.storage.get()) + static_cast<int>(p.piece); }
```

Comment:
pretty sure this generate a double conversion, first to `signed` for the `+` and then to `unsigned` at return